src/pages/Product.tsx

```diff
@@ -3,17 +3,46 @@ import * as React from 'react';
 import { ProductInformation } from '../sections/';
 import { AllProductDescription, SimpleBreadcrumbs } from '../components';
 
-function Product() {
+import { fetchJewelryById } from '../redux/reducers/jewelry/action';
+import { IJewelryItem } from '../redux/reducers/jewelry/type';
+
+import * as H from 'history';
+
+interface Product {
+  id: any;
```

Comment:
Avoid using type any. 
Use a union:
`id: number | string;`
https://betterprogramming.pub/top-5-typescript-features-you-should-master-2358db9ab3d5